src/visualization/makie_visualization.jl

```diff
@@ -312,13 +335,35 @@ Makie.plottype(::Trixi.PlotDataSeries2D{<:Trixi.UnstructuredPlotData2D}) = Trixi
 # Makie does not yet support layouts in its plot recipes, so we overload `Makie.plot` directly.
 function Makie.plot(sol::TrixiODESolution;
                     plot_mesh=true, solution_variables=nothing)
-  Makie.plot(PlotData2D(sol, solution_variables=solution_variables); plot_mesh=plot_mesh)
+  return Makie.plot(PlotData2D(sol, solution_variables=solution_variables); plot_mesh=plot_mesh)
 end
 
-function Makie.plot(pd::UnstructuredPlotData2D, fig = Makie.Figure();
-                    plot_mesh = true)
-  Makie.plot!(fig, pd; plot_mesh=plot_mesh)
-  fig
+# convenience struct for editing Makie plots after they're created.
+struct FigureAndAxes{Axes}
+  fig::Makie.Figure
+  axes::Axes
+end
+
+# for "quiet" return arguments to Makie.plot(::TrixiODESolution) and
+# Makie.plot(::UnstructuredPlotData2D)
+Base.show(io::IO, fa::FigureAndAxes) = nothing
+
+# allows for returning fig, axes = Makie.plot(...)
+function Base.iterate(fa::FigureAndAxes, state=1)
+  if state > 2
+    return nothing
+  elseif state==1
+    return (fa.fig, 2)
+  elseif state==2
+    return (fa.axes, 3)
+  end
```

Comment:
```suggestion
  if state == 1
    return (fa.fig, 2)
  elseif state == 2
    return (fa.axes, 3)
  else
    return nothing
  end
```
This also handles `state < 1` (which might happen by accident if a user calls `iterate` explicitly with wrong `state`) and removes this branch, making it easier for the compiler.